worlds/mmbn3/__init__.py

```diff
@@ -0,0 +1,486 @@
+                entrance.connect(connection_region)
+
+                # ACDC Pending with Start Randomizer
+                # if connection == RegionName.ACDC_Overworld:
+                #     entrance.access_rule = lambda state: state.has(ItemName.Parasol, self.player)
+                if connection == RegionName.SciLab_Overworld:
+                    entrance.access_rule = lambda state: state.has(ItemName.SubPET, self.player)
+                if connection == RegionName.Yoka_Overworld:
+                    entrance.access_rule = lambda state: state.has(ItemName.Needle, self.player)
+                if connection == RegionName.Beach_Overworld:
+                    entrance.access_rule = lambda state: state.has(ItemName.PETCase, self.player)
+
+                # ACDC Pending with Start Randomizer
+                # if connection == RegionName.ACDC_Cyberworld:
+                #     entrance.access_rule = lambda state: state.has(ItemName.CACDCPas, self.player)
+                if connection == RegionName.SciLab_Cyberworld:
+                    entrance.access_rule = lambda state: \
+                        state.has(ItemName.CSciPas, self.player) or \
+                        state.can_reach(RegionName.SciLab_Overworld, "Region", self.player)
+                if connection == RegionName.Yoka_Cyberworld:
+                    entrance.access_rule = lambda state: \
+                        state.has(ItemName.CYokaPas, self.player) or \
+                        (
+                            state.can_reach(RegionName.SciLab_Overworld, "Region", self.player) and
+                            state.has(ItemName.Press, self.player)
+                        )
+                if connection == RegionName.Beach_Cyberworld:
+                    entrance.access_rule = lambda state: state.has(ItemName.CBeacPas, self.player) and\
+                        state.can_reach(RegionName.Yoka_Overworld, "Region", self.player)
+
+                if connection == RegionName.Undernet:
+                    entrance.access_rule = lambda state: self.explore_score(state) > 8 and\
+                        state.has(ItemName.Press, self.player)
+                if connection == RegionName.Secret_Area:
+                    entrance.access_rule = lambda state: self.explore_score(state) > 12 and\
+                        state.has(ItemName.Hammer, self.player)
+                if connection == RegionName.WWW_Island:
+                    entrance.access_rule = lambda state:\
+                        state.has(ItemName.Progressive_Undernet_Rank, self.player, 8)
+                region.exits.append(entrance)
+
+    def create_items(self) -> None:
+        # First add in all progression and useful items
+        required_items = []
+        for item in all_items:
+            if item.progression != ItemClassification.filler:
+                freq = self.item_frequencies.get(item.itemName, 1)
+                required_items += [item.itemName] * freq
```

Comment:
Do not use this kind of syntax for creating multiple items. This ends up duplicating the same reference rather than creating a new instance. It might not be terrible in your use case, but I'd recommend this instead just to be in the habit:

```suggestion
                required_items += [item.itemName for _ in range(freq)]
```